test/jasmine/tests/heatmap_test.js

```diff
@@ -404,6 +409,25 @@ describe('heatmap calc', function() {
         expect(out.y).toBeCloseToArray([-0.5, 0.5]);
         expect(out.z).toBeCloseTo2DArray([[17, 18, 19]]);
     });
+
+    it('should handle the category x/y/z/ column case', function() {
+        var out = _calc({
+            x: ['a', 'a', 'a', 'b', 'b', 'b', 'c', 'c', 'c'],
+            y: ['A', 'B', 'C', 'A', 'B', 'C', 'A', 'B', 'C'],
```

Comment:
can we test dates too? That's the other case where `d2c` shouldn't be called multiple times - actually, that results in a timezone shift because of our legacy support for local milliseconds!
```
ax.d2c('1970-01-02')
86400000
ax.d2c(ax.d2c('1970-01-02'))
68400000
ax.d2c(ax.d2c(ax.d2c('1970-01-02')))
50400000
```